certificate-service/app/src/main/java/se/inera/intyg/certificateservice/application/certificate/service/CertificateMetaDataUnitConverter.java

```diff
@@ -0,0 +1,39 @@
+package se.inera.intyg.certificateservice.application.certificate.service;
+
+import static se.inera.intyg.certificateservice.application.common.CertificateConstants.ISSUING_UNIT;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.stereotype.Component;
+import se.inera.intyg.certificateservice.application.certificate.dto.UnitDTO;
+import se.inera.intyg.certificateservice.domain.certificate.model.ElementData;
+import se.inera.intyg.certificateservice.domain.certificate.model.ElementValueIssuingUnit;
+import se.inera.intyg.certificateservice.domain.certificate.model.IssuingUnit;
+import se.inera.intyg.certificateservice.domain.certificatemodel.model.ElementId;
+
+@Component
+public class CertificateMetaDataUnitConverter {
+
+
+  public UnitDTO convert(IssuingUnit issuingUnit, List<ElementData> elementData) {
```

Comment:
See comment on the class calling this. If elementData can be an optional you can simplify the code and just do a .map(...) to create a UnitDTO from the elementData orElse(...) create it from the IssuingUnit.

Reply:
Thats brilliant, great suggestion!